### PR TITLE
pull ollama models if not exist automatically

### PR DESCRIPTION
**Pull request by cyyeh**
if users choose Ollama models, ai service would check first if they are available, and pull them automatically if not available.

### CODE REVIEW

File: wren-ai-service/src/providers/engine/wren.py [hunk 1]
```diff
@@ -54,8 +55,10 @@ async def dry_run_sql(
             "source": os.getenv("WREN_IBIS_SOURCE"),
             "manifest": os.getenv("WREN_IBIS_MANIFEST"),
             "connection_info": orjson.loads(
-                os.getenv("WREN_IBIS_CONNECTION_INFO", "{}")
-            ),
+                base64.b64decode(os.getenv("WREN_IBIS_CONNECTION_INFO"))
+            )
+            if os.getenv("WREN_IBIS_CONNECTION_INFO")
+            else {},
```
Comment (paopa) [inline wren-ai-service/src/providers/engine/wren.py:61]: could you modify the example for env.dev file as well if we'd like to encode the connection info

File: wren-ai-service/src/providers/loader.py [hunk 2]
```diff
@@ -98,3 +100,19 @@ def get_default_embedding_model_dim(embedder_provider: str):
...
+    models = client.list()["models"]
+    if model_name not in models:
+        logger.info(f"Pulling Ollama model {model_name}")
+        percentage = 0
+        for progress in client.pull(model_name, stream=True):
+            if "completed" in progress and "total" in progress:
+                new_percentage = int(progress["completed"] / progress["total"] * 100)
+                if new_percentage > percentage:
+                    percentage = new_percentage
+                    logger.info(f"Pulling Ollama model {model_name}: {percentage}%")
```
[the remark this answers is not in the document]
Comment (paopa) [reply wren-ai-service/src/providers/loader.py:116]: have discussed offline.

File: wren-ai-service/src/providers/engine/wren.py (hunk 1 above)
Comment (paopa) [reply wren-ai-service/src/providers/engine/wren.py:61]: have discussed offline

**Review by paopa — APPROVED** [review]
LGTM

**Review by onlyjackfrost — APPROVED** [review]

**Pull request merged**